Doc/using/cmdline.rst

```diff
@@ -994,3 +1000,110 @@ Debug-mode variables
+directory, which is the case when executing the interactive shell, a :ref:`-c <using-on-interface-option-c>`
+command, or :ref:`-m <using-on-interface-option-m>` module.
+
+The :envvar:`PYTHONPATH` environment variable is often used to add directories
+to the search path. If this environment variable is found then the contents are
+added to the module search path.
+
+.. note::
+
+   :envvar:`PYTHONPATH` will affect all installed Python versions/environments.
+   Be wary of setting this in your shell profile or global environment variables.
+   The :mod:`site` module offers more nuanced techniques as mentioned below.
+
+The next items added are the directories containing standard Python modules as
+well as any :term:`extension module`\s that these modules depend on. Extension
+modules are .pyd files on Windows and .so files on other platforms. The
+directory with the platform-independent Python modules is called ``prefix``.
+The directory with the extension modules is called ``exec_prefix``.
+
+The :envvar:`PYTHONHOME` environment variable may be used to set the ``prefix``
+and ``exec_prefix`` locations. Otherwise these directories are found by using
+the Python executable as a starting point and then looking for various 'landmark'
+files and directories. Note that any symbolic links are followed so the real
+Python executable location is used as the search starting point. The Python
+executable location is called ``home``.
+
+Once ``home`` is determined, the ``prefix`` directory is found by first looking
+for :file:`python{majorversion}{minorversion}.zip` (``python311.zip``). On Windows
+the zip archive is searched for in ``home`` and on Unix the archive is expected
+to be in :file:`lib`. Note that the expected zip archive location is added to the
+module search path even if the archive does not exist. If no archive was found,
+Python on Windows will continue the search for ``prefix`` by looking for :file:`Lib\\os.py`.
+Python on Unix will look for :file:`lib/python{majorversion}.{minorversion}/os.py`
+(``lib/python3.11/os.py``). On Windows ``prefix`` and ``exec_prefix`` are the same,
+however on other platforms :file:`lib/python{majorversion}.{minorversion}/lib-dynload`
+(``lib/python3.11/lib-dynload``) is searched for and used as an anchor for
+``exec_prefix``.
+
+Once found, ``prefix`` and ``exec_prefix`` may be accessed at :data:`sys.prefix`
+and :data:`sys.exec_prefix`.
+
+Finally, the :mod:`site` module is processed and :file:`site-packages` directories
+are added to the module search path. A common way to customize the search path is
+to create :mod:`sitecustomize` or :mod:`usercustomize` modules as described in
+the :mod:`site` module documentation.
+
+.. note::
+
+   Certain command line options may further affect path calculations.
+   See :ref:`-E <using-on-misc-option-uppercase-e>`, :ref:`-I <using-on-misc-option-uppercase-i>`, :ref:`-s <using-on-misc-option-s>` and :ref:`-S <using-on-misc-option-uppercase-s>` for further details.
```

Comment:
In terms of formatting, I suggest a list instead:

```suggestion
Certain command line options may further affect path calculations:

* :ref:`-E <using-on-misc-option-uppercase-e>`: ignore environment variables
* :ref:`-I <using-on-misc-option-uppercase-i>`: isolated mode
* :ref:`-s <using-on-misc-option-s>`: don't add user directory
* :ref:`-S <using-on-misc-option-uppercase-s>`: don't import the :mod:`site` module
```

It would be nice to have at one place, the exhaustive list of things impacting sys.path:

* current working directory
* path of the Python executable program
* list of cmdline options
* list of env vars
* list of config files: see my list at https://docs.python.org/dev/c-api/init_config.html#python-path-configuration

Reply:
I had a more verbose description, see https://github.com/python/cpython/pull/31082/commits/59cbc07c783610c8f9204b8a5fed4881f3995a82#r822648316 however @zooba preferred the more succint approach.